Replace model parameter default schema versions by value read from model_parameter.metaschema.yml

GernotMaier opened the pull request:
Addresses the repetition of getting / reading the model parameters schema files and removes the necessity of detailed knowledge about paths and naming of schema files (especially for those in `src/simtools/schemas/model_parameters/`):

- introduces a now module `simtools.data_model.schema` (recommend to start reviewing that)
- add several of the main schema information (paths and files) to `simtools.constants`
- removes / changes wherever possible the code related to schema files to use the generalized approach
- add a function to get the most recent schema version for the model parameters (or check that a given version actually exists): `schema.model_parameter_schema_version`. Closes #1327 
- add a small improvement to `simtools.utils.general.collect_data_from_file` to get the Nth document of a multi-document yaml file

GernotMaier commented:
Just did a merge from the branch `db-simulation-model-refactoring` - this should fix the failing integration tests.

tobiaskleiner commented on src/simtools/data_model/schema.py:
```diff
@@ -0,0 +1,77 @@
+from simtools.constants import MODEL_PARAMETER_METASCHEMA, MODEL_PARAMETER_SCHEMA_PATH
+
+
+def model_parameter_schema_files(schema_directory=MODEL_PARAMETER_SCHEMA_PATH):
```
Suggest to rename the functions to make their purpose clearer. Maybe `get_parameters_and_schema_files`, also the following `get_schema_path_for_model_parameter`and `get_validated_schema_version`.

GernotMaier replied:
OK - for me it appears clear the `schema.model_parameter_schema_files` gives me the schema files. But your comment shows that this is not the case for everyone, so I changed it and added everywhere the `get_`...

orelgueta commented:
> **Note: merges into [db-simulation-model-refactoring](https://github.com/gammasim/simtools/tree/db-simulation-model-refactoring) branch - the 3 integration tests fails until PR #1319 is reviewed an merged**
 
Is this still correct?

GernotMaier commented:
> > **Note: merges into [db-simulation-model-refactoring](https://github.com/gammasim/simtools/tree/db-simulation-model-refactoring) branch - the 3 integration tests fails until PR #1319 is reviewed an merged**
> 
> Is this still correct?

No. Removed it.

tobiaskleiner reviewed (commented):
Thanks @GernotMaier, looks good overall. Added a minor comment.

GernotMaier commented:
@tobiaskleiner - thanks for the review. I've addressed the comment and changed the function names in `simtools.data_model.schema`. let me know if there is anything else.
